mediagrains/comparison/_internal.py

```diff
@@ -542,8 +603,204 @@ def compare(self, a, b):
 class GrainComparisonResult(ComparisonResult):
     """A ComparisonResult class for comparing grains, this is where almost all of the grain comparison logic is
     contained."""
-    def compare(self, a, b):
-        children = {}
+    def compare_event_grains(self, a: EVENTGRAIN, b: EVENTGRAIN, children) -> None:
+        # We are comparing event grains, so compare their event grain specific features
+        for key in ['event_type',
+                    'topic']:
+            path = self._identifier + '.' + key
+            children[key] = EqualityComparisonResult(
+                path, getattr(a, key), getattr(b, key), options=self._options, attr=key)
+        for key in ['event_data']:
+            path = self._identifier + '.' + key
+            children[key] = OrderedContainerComparisonResult(
+                self._identifier + '.' + key,
+                getattr(a, key),
+                getattr(b, key),
+                options=self._options,
+                comparison_class=MappingContainerComparisonResult,
+                attr=key)
+        if children['event_type'].excluded() or children['topic'].excluded() or children['event_data'].excluded():
+            children['length']._options.append(Exclude.length)
+
+    def compare_audio_grains(self, a: AUDIOGRAIN, b: AUDIOGRAIN, children) -> None:
+        # We are comparing audio grains, so compare their audio grain specific features
+        for key in ['format',
+                    'samples',
+                    'channels',
+                    'sample_rate']:
+            path = self._identifier + '.' + key
+            children[key] = EqualityComparisonResult(
+                path, getattr(a, key), getattr(b, key), options=self._options, attr=key)
+
+        if a.format == b.format:
+            path = self._identifier + '.data'
+            compare_psnr = len(
+                [option for option in self._options if isinstance(option, ComparisonPSNR) and path == option.path]
+                ) != 0
+            if compare_psnr:
+                children['data'] = PSNRComparisonResult(path,
+                                                        a,
+                                                        b,
+                                                        options=self._options)
+            else:
+                wps = 1
```

Comment:
Nit: Would you mind renaming `wps` to `words_per_sample`, `wc` to `word_code` and `s` to `force_signed`? It would make this a bit easier to follow than the original I think!